src/pybind/mgr/volumes/fs/misc_util.py

```diff
@@ -0,0 +1,206 @@
+import errno
+import stat
+from os.path import join as os_path_join
+from logging import getLogger
+
+
+import cephfs
+from .exception import VolumeException
+
+
+log = getLogger(__name__)
+
+
+def get_size_h(num):
+    '''
+    Convert size to human-readable format.
+    '''
+    size = ('B', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB')
+
+    if num < 1024:
+        return f'{num} {size[0]}'
+
+    i = 0
+    while True:
+        num = num / 1024
+        i += 1
+        if i >= len(size):
+            log.error(f"Biggest suffix we have is '{len(size)-1}' "
+                       'but this is not sufficient in this case. '
+                      f'Current size = {num}')
+
+        x, y = str(num).split('.')
+        if len(x) <= 3:
+            if y:
+                y = '' if y == '0' else y[:4]
+
+            return f'{x}.{y} {size[i]}' if y else f'{x} {size[i]}'
+
+
+def get_num_h(num):
+    '''
+    Convert number to human-readable format.
+    '''
+    size = ('', 'K', 'M', 'B', 'T')
+
+    if num < 1000:
+        return f'{num}'
+
+    i = 0
+    while True:
+        num = num / 1000
+        i += 1
+        if i >= len(size):
+            log.error(f"Biggest suffix we have is '{len(size)-1}' "
+                       'but this is not sufficient in this case. '
+                      f'Current size = {num}')
+
+        x, y = str(num).split('.')
+        if len(x) <= 3:
+            if y:
+                y = '' if y == '0' else y[:4]
+
+            return f'{x}.{y} {size[i]}' if y else f'{x} {size[i]}'
+
+
+class Stats:
+    '''
+    Store statistics of the given path that'll be copied/deleted/purged (or
+    handled, in short) in an organized manner.
+    '''
+    def __init__(self, op):
+        self.size_t = 0     # total size
+        self.rfiles_t = 0   # total regfiles
+        self.dirs_t = 0     # total directories
+        self.slinks_t = 0   # total symlinks
+
+        # handled = copied, delete/purged, etc.
+        self.size_h = 0     # size handled
+        self.rfiles_h = 0   # num of regfiles handled
+        self.dirs_h = 0     # num of dirs handled
+        self.slinks_h = 0   # num of symlinks handled
+
+        self.op = op
+        # creating self.oped so that we can pring log messages accordingly.
+        self.oped = self._get_oped_str(op) # oped = operated
+
+    @property
+    def precent(self):
+        if self.size_t == 0:
+            return 100 if self.size_h == 0 else 0
+        return ((self.size_h/self.size_t) * 100)
+
+    def _get_oped_str(self, op):
```

Comment:
Rather than introducing calls for each file type, have a single callable that is passed a bitmap (OR'd) of items (types) to be fetched (like statx).